### ISSUE TITLE
'NoneType' object has no attribute 'encode'

### ISSUE THREAD

**Issue by zhbzhbzhbz** [miurahr/py7zr#215]
[test.zip](https://github.com/miurahr/py7zr/files/5050237/test.zip)
**How the bug occured**
executing "SevenZipFile" on a encrypted 7z file.

**To Reproduce**
run "py7zr.SevenZipFile("test.7z", mode='r')" then the error occurs:

 ```
File "F:\Forum Spider\core\decompress\decompress_utils.py", line 108, in un7z
    with py7zr.SevenZipFile(filepath, mode='r') as archive:
  File "D:\Program Files\Python38\lib\site-packages\py7zr\py7zr.py", line 345, in __init__
    raise e
  File "D:\Program Files\Python38\lib\site-packages\py7zr\py7zr.py", line 318, in __init__
    self._real_get_contents(password)
  File "D:\Program Files\Python38\lib\site-packages\py7zr\py7zr.py", line 373, in _real_get_contents
    header = Header.retrieve(self.fp, buffer, self.afterheader, password)
  File "D:\Program Files\Python38\lib\site-packages\py7zr\archiveinfo.py", line 850, in retrieve
    obj._read(fp, buffer, start_pos, password)
  File "D:\Program Files\Python38\lib\site-packages\py7zr\archiveinfo.py", line 883, in _read
    decompressor = folder.get_decompressor(compressed_size)
  File "D:\Program Files\Python38\lib\site-packages\py7zr\archiveinfo.py", line 390, in get_decompressor
    self.decompressor = SevenZipDecompressor(self.coders, packsize, self.unpacksizes, self.crc, self.password)
  File "D:\Program Files\Python38\lib\site-packages\py7zr\compressor.py", line 744, in __init__
    self.cchain.add_filter(get_alternative_decompressor(coders[j], unpacksizes[j], password))
  File "D:\Program Files\Python38\lib\site-packages\py7zr\compressor.py", line 600, in get_alternative_decompressor
    return algorithm_class_map[filter_id][1](coder['properties'], password)
  File "D:\Program Files\Python38\lib\site-packages\py7zr\compressor.py", line 168, in __init__
    key = calculate_key(password.encode('utf-16LE'), numcyclespower, salt, 'sha256')
AttributeError: 'NoneType' object has no attribute 'encode'
```

**Environment (please complete the following information):**
 - OS: Windows 10
 - Python 3.8.5
 - py7zr version: 0.9.3

**Comment by miurahr** [comment]
What expected without password?

**Comment by zhbzhbzhbz** [comment]
> What expected without password?

password is 123456

You mean exception's name? "PasswordRequired" or "RarWrongPassword" like rarfile?

**Comment by miurahr** [comment]
Does #216 behave as your expected, @zhbzhbzhbz ?

**Comment by zhbzhbzhbz** [comment]
> Does #216 behave as your expected, @zhbzhbzhbz ?

Uh, It looks right.

**Comment by miurahr** [comment]
Fix is merged into master branch and releases/0.9, push v0.9.4 soon.
